packages/cache/src/lib/CacheMiddleware.ts

```diff
@@ -0,0 +1,672 @@
+
+    if (entries) {
+      payload.data ??= {};
+
+      for (const [key, val] of Object.entries(entries)) {
+        if (await this.checkNotExpired(val, key)) {
+          payload.data[key] = val.value;
+        } else {
+          const bypassPayload = await this.provider[Method.Get]({ method: Method.Get, key, path: [], errors: [] });
+
+          if (isPayloadWithData<StoredValue>(bypassPayload)) {
+            const { data } = bypassPayload;
+
+            payload.data[key] = data;
+          }
+        }
+      }
+    }
+
+    return payload;
+  }
+
+  public async [Method.Every](payload: Payloads.Every.ByHook<StoredValue>): Promise<Payloads.Every.ByHook<StoredValue>>;
+  public async [Method.Every](payload: Payloads.Every.ByValue): Promise<Payloads.Every.ByValue>;
+
+  @PreProvider()
+  public async [Method.Every](payload: Payloads.Every<StoredValue>): Promise<Payloads.Every<StoredValue>> {
+    const { provider: cache } = this.context;
+    const { type, hook, path, value } = payload;
+    const cacheHook = (hook: Payload.Hook<StoredValue, unknown>) => {
+      return async (value: CacheMiddleware.Document<StoredValue>, key: string) => {
+        if (await this.checkNotExpired(value, key)) {
+          return hook(value.value, key);
+        }
+
+        const bypassPayload = await this.provider[Method.Get]({ method: Method.Get, key, path: [], errors: [] });
+
+        if (isPayloadWithData<StoredValue>(bypassPayload)) {
+          const { data } = bypassPayload;
+
+          return hook(data, key);
+        }
+
+        return true; // I think(?) this is ok
+      };
+    };
+
+    if (isEveryByHookPayload(payload) && hook) {
+      const { data, errors } = await cache[Method.Every]({ method: Method.Every, type, data: true, hook: cacheHook(hook), path, errors: [] });
+
+      payload.data = data;
+      payload.errors = [...payload.errors, ...errors];
+    } else {
+      const valueHook = (storedValue: StoredValue) => {
+        const data = getProperty(storedValue, path || []);
+
+        return data === value;
+      };
+
+      const { data, errors } = await cache[Method.Every]({
+        method: Method.Every,
+        type: Payload.Type.Hook,
+        data: true,
+        hook: cacheHook(valueHook),
+        path,
+        errors: []
+      });
+
+      payload.data = data;
+      payload.errors = [...payload.errors, ...errors];
+    }
+
+    return payload;
+  }
+
+  public async [Method.Filter](payload: Payloads.Filter.ByHook<StoredValue>): Promise<Payloads.Filter.ByHook<StoredValue>>;
+  public async [Method.Filter](payload: Payloads.Filter.ByValue<StoredValue>): Promise<Payloads.Filter.ByValue<StoredValue>>;
+
+  @PreProvider()
+  public async [Method.Filter](payload: Payloads.Filter<StoredValue>): Promise<Payloads.Filter<StoredValue>> {
+    const { provider: cache } = this.context;
+    const { hook, path, type, value } = payload;
+
+    payload.data = {};
+
+    if (hook && isFilterByHookPayload(payload)) {
```

Comment:
```suggestion
    if (isFilterByHookPayload(payload)) {
```
Move the destructure of `hook` inside this scope